### PR TITLE
Implement GetAvailableLanguageCodes2

### PR DESCRIPTION
**Pull request by greggameplayer**
tested on Kirby Star Allies

### CODE REVIEW

File: src/core/hle/service/set/set.cpp [hunk 1]
```diff
@@ -43,14 +43,46 @@ void SET::GetAvailableLanguageCodes(Kernel::HLERequestContext& ctx) {
     NGLOG_DEBUG(Service_SET, "called");
 }
 
+void SET::GetAvailableLanguageCodes2(Kernel::HLERequestContext& ctx) {
+    IPC::RequestParser rp{ctx};
+    u32 id = rp.Pop<u32>();
```
Comment (mailwl) [inline src/core/hle/service/set/set.cpp]: unused, maybe log this?

Comment (FearlessTobi) [reply src/core/hle/service/set/set.cpp]: This is basically a copy of the GetAvailableLanguageCodes implementation and the value doesn't get logged there. So if it should be changed, then in both functions.

Comment (greggameplayer) [reply src/core/hle/service/set/set.cpp]: ok so i delete this in both functions

File: src/core/hle/service/set/set.cpp [hunk 2]
```diff
@@ -12,9 +12,35 @@
...
+    ctx.WriteBuffer(available_language_codes.data(), available_language_codes.size());
+
+    IPC::ResponseBuilder rb{ctx, 4};
+    rb.Push(RESULT_SUCCESS);
+    rb.Push(static_cast<u64>(available_language_codes.size()));
+
+    NGLOG_DEBUG(Service_SET, "called");
+}
 
+void SET::GetAvailableLanguageCodes2(Kernel::HLERequestContext& ctx) {
```
Comment (bunnei) [inline src/core/hle/service/set/set.cpp]: This is just a copy paste of `GetAvailableLanguageCodes` .. just call that directly if it's the same function

Comment (greggameplayer) [reply src/core/hle/service/set/set.cpp]: ok just the buffer type is different

**Comment by greggameplayer** [comment]
bug in squash

**Comment by greggameplayer** [comment]
@bunnei done

**Pull request merged**